File: Seori/LeetCode 75/Array_String/1768. Merge Strings Alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        # word1부터 하나씩 꺼내서 merged_word를 만들면서, 각 word의 길이보다 작은 경우에만 추가한다.
        # From word1, take one by one and make merged_word, only if the length is less than each word
        merged_word = ''
        len1, len2 = len(word1), len(word2)
        length = max(len1, len2)
        for i in range(length):
            if i < len1:
                merged_word += word1[i]
            if i < len2:
                merged_word += word2[i]
        
        return merged_word

File: Gombang/[LeetCode75] - [Array_Sort]/1768_Merge_Strings_Alternately.cs
public class Solution
{
	public string MergeAlternately(string word1, string word2)
	{
		int strIndex = 0;
		StringBuilder str = new StringBuilder();

		while (strIndex < word1.Length && strIndex < word2.Length)
		{
			str.Append(word1[strIndex]);
			str.Append(word2[strIndex]);

			strIndex++;
		}

		if (strIndex < word1.Length)
		{
			str.Append(word1.Substring(strIndex, word1.Length - strIndex));
		}
		else if (strIndex < word2.Length)
		{
			str.Append(word2.Substring(strIndex, word2.Length - strIndex));
		}

		return str.ToString();
	}
}